include/igl/marching_tets.cpp

```diff
@@ -151,6 +151,10 @@ void igl::marching_tets(
     {
       const int vi = faces[f].first[v];
       const int ti = faces[f].second;
+      assert(ti>=0);
+      assert(ti<TT.rows());
+      // this will overwrite, but should be with the same value
+      J(f) = ti;
```

Comment:
We could also move the `const int ti = faces[f].second;` outside the loop on `v`.